scripts/release.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+# release.sh
+#
+# This script automates the process of preparing a new release for mirrord.
+#
+# Usage:
+#   ./release.sh 1.234.5
+#
+# Requirements:
+# - `git`
+# - `cargo-set` for setting workspace package version
+# - `towncrier` for changelog generation
+#
+# What this script does:
+# - Checkout a new branch and use the release version as the name of the branch.
+# - Bump workspace version in `Cargo.toml`.
+# - Prepare changelog using `towncrier`.
+# - Push the branch to remote.
+#
+# Continue with the following steps after running this script:
+# - Create a PR and merge it into mirrord main.
+# - Create a new Github release with a new tag. This triggers the `Release` workflow.
+#     - Use the new version for the tag and relase name.
+#     - Use the changelog as the release description excluding the `Internal` section.
+# - Once the workflow finishes, adjust environment variables in our analytics server.
+#
+
+set -e
+
+# Print usage and exit
+usage() {
+    echo "Usage: $0 <version>"
+    echo "Example: $0 1.234.5"
+    exit 1
+}
+
+# Check for version argument
+VERSION="$1"
+if [[ ! "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo "Error: Invalid version format: '$VERSION'"
+    usage
+fi
+
+# Ensure we are in the repo root
+cd "$(git rev-parse --show-toplevel)"
+
+# Ensure we're on the main branch or switch to it
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+if [[ "$CURRENT_BRANCH" != "main" ]]; then
+    if [[ -n $(git status --porcelain) ]]; then
+        echo "Error: You are on branch '$CURRENT_BRANCH' with uncommitted or unstaged changes. Please commit or stash them before switching to 'main'."
+        exit 1
+    else
+        echo "You are on branch '$CURRENT_BRANCH'. Switching to 'main'..."
+        git checkout main
+    fi
+fi
+
+# Ensure main is up to date
+git pull origin main
```

Comment:
Some devs might call the remote something else, like e.g. "upstream", and I think that's valid (even though mine is called "origin" like yours). Is there a reason not to change this to
```suggestion
git pull
```
since we're already on `main`?

Reply:
Changed this to
1. prompt the user for their choice of remote by providing a list of available remotes in the local repo
2. default to origin
3. `git pull "$selected_remote" main` in case the selected remote is different from the remote that main is tracking.

Reply:
Why not just `git pull` though?